src/libertem/analysis/base.py

```diff
@@ -172,6 +172,22 @@ class Analysis:
     # https://www.python.org/dev/peps/pep-0586/
     TYPE: typing.Union[str, None] = None
 
+    registry = {}
+
+    def __init_subclass__(cls, type_=None, **kwargs):
```

Comment:
Hmm, can we rename `type_`? Like this, there are two "types" for an analysis, this one and the UDF/JOB one. Maybe `id_` or `name` also fits?

Reply:
will go with `id_`